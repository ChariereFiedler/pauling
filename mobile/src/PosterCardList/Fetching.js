import React from 'react';

import { View } from 'react-native';
import { Spinner, Text } from 'native-base';

import { colors } from '../settings';
import styles from './styles';

const Fetching = () =>
  <View style={styles.Fetching}>
    <Text style={styles.FetchingMessage}>
      Collecting poster from Pauling databaseâ€¦
    </Text>
    <Spinner color={colors.textAltPrimaryColor} />
  </View>;

export default Fetching;
