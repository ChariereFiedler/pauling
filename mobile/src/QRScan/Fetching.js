import React, { Component } from 'react';

import { View } from 'react-native';
import { Spinner, Text } from 'native-base';

import { colors } from '../settings';
import type { Poster } from '../types';
import styles from './styles';

type Props = {};


class Fetching extends Component {

  props: Props;

  constructor(props: Props) {
    super(props);

    this.state = {
      hasFetchedData: false,
    };
  };

  render() {
    return (
      <View style={styles.Fetching}>
        <Text style={styles.FetchingMessage}>
          Collecting poster from Pauling databaseâ€¦
        </Text>
        <Spinner color={colors.textAltPrimaryColor} />
      </View>
    );
  }
}

export default Fetching;
