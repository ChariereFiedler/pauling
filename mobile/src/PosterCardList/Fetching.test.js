import React from 'react';
import { shallow } from 'enzyme';

import Fetching from 'app/PosterCardList/Fetching';

describe('Fetching', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Fetching />);
    expect(wrapper).toMatchSnapshot();
  });
});
