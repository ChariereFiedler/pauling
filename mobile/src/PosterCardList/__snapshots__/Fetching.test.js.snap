// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetching renders correctly 1`] = `
<Component
  style={Object {}}
>
  <Styled(Text)
    style={
      Object {
        "color": "#777",
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 50,
        "textAlign": "center",
      }
    }
  >
    Fetching the poster from Pauling, this should not take too long (in theory).
  </Styled(Text)>
  <Styled(Spinner)
    color="#777"
  />
</Component>
`;
